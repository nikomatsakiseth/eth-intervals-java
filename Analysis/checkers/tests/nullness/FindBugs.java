import edu.umd.cs.findbugs.annotations.*;

public class FindBugs {

    @CheckForNull Object getNull() {
        return null;
    }

    @NonNull MyList<@Nullable Object> getListOfNulls() {
        return null;    // error
    }

    void test() {
        Object o = getNull();
        o.toString();   // error

        MyList<@Nullable Object> l = getListOfNulls();
        l.toString();
        l.get().toString();    // error
    }
}

class MyList<T extends @Nullable Object> {
    T get() { throw new RuntimeException(); }
}