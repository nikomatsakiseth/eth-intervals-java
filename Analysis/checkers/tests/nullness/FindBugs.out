:10: (type.incompatible)
:15: (dereference.of.nullable)
:19: (dereference.of.nullable)
